server/src/main/java/org/candlepin/resteasy/parameter/KeyValueParameter.java

```diff
@@ -31,30 +31,24 @@
      *     FORMAT: paramName=key:value
      * </pre>
      *
-     * @param queryParameterValue the value of the query parameter as entered in the URL
+     * @param inValue the value of the query parameter as entered in the URL
      *                            in the format: key:value
      */
-    public KeyValueParameter(String queryParameterValue) {
-        super(queryParameterValue);
-    }
-
-    @Override
-    void parse() throws CandlepinParameterParseException {
-        String[] parts = this.paramValue.split(":", 2); // Maximum of two parts
+    public KeyValueParameter(String inValue) {
+        String[] parts = inValue.split(":", 2); // Maximum of two parts
         if (parts.length == 1) {
-            throw new CandlepinParameterParseException(this.paramName,
-                this.paramName + "=name:value");
+            throw new CandlepinParameterParseException("name:value");
```

Comment:
Looks like the price of switching KeyValueParameter to just being a regular, bindable class is that we don't have the parameter name anymore to show in an error message.  E.g. "Bad format; Use fact=name:value" becomes "Bad format; Use name:value".  I'm okay with that, but just wanted to make sure that was a conscious choice.

Reply:
@awood yes, this was done consciously.